lib/vrl/core/src/runtime.rs

```diff
@@ -52,6 +52,7 @@ impl Runtime {
         target: &mut dyn Target,
         program: &Program,
         timezone: &TimeZone,
+        enrichment_tables: &Option<Box<dyn EnrichmentTableSearch>>,
```

Comment:
```suggestion
        enrichment_tables: Option<&Box<dyn EnrichmentTableSearch>>,
```

Reply:
If I'm understanding the logic here correctly, this can potentially be misused.

That is, what if I pass in a different list of enrichment tables to `compile` and `resolve`? The compiler wouldn't error if you'd reference an enrichment table that isn't passed in to `resolve`, right?

Ideally, the compiled `Program` keeps a reference to the enrichment tables it needs access to, such that the runtime is guaranteed to have access to it as well.

Reply:
Good call.